### PR TITLE
Add constraints columns to table system.settings

### PR DESCRIPTION
**Pull request by vitlibar**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Add new columns `min`, `max`, `readonly` to the `system.settings` table.

Category (leave one):
- Improvement

### CODE REVIEW

File: dbms/src/Storages/System/StorageSystemSettings.cpp [hunk 1]
```diff
@@ -1,11 +1,25 @@
...
+#include <Access/SettingsConstraints.h>
 
 
 namespace DB
 {
+String StorageSystemSettings::valueToString(const StringRef & name, const Field & value)
+{
+    try
+    {
+        return Settings::getAsStringWithoutApplying(name, value);
```
Comment (alexey-milovidov) [inline dbms/src/Storages/System/StorageSystemSettings.cpp]: Why it can throw exception?

Comment (vitlibar) [reply dbms/src/Storages/System/StorageSystemSettings.cpp]: You're right, it shouldn't throw exceptions. I've changed this code.

**Comment by BayoNet** [comment]
@vitlibar 
min - Minimum value of setting.
max - Maximum value of settings.
readonly - Flag showing that the setting can't be changed by user.

Am I right?

If I'm right, I want to know why some settings in this table can't be changed? Aren't there are only session settings that can be changed with the SET query?

**Comment by vitlibar** [comment]
> @vitlibar
> min - Minimum value of setting.
> max - Maximum value of setting.
> readonly - Flag showing that the setting can't be changed by user.
> 
> Am I right?

Yes.

> If I'm right, I want to know why some settings in this table can't be changed?

Maybe it's because the setting `readonly` is set:

```
SET readonly=1; -- set the `readonly` constraint for all settings
```

Or maybe it's because `users.xml` prohibit changing this setting (see [Constraints on Settings](https://clickhouse.yandex/docs/en/operations/settings/constraints_on_settings/)).

> Aren't there are only session settings that can be changed with the SET query?

The SET query changes session settings. If your change violates some constrains an exception is thrown.

**Pull request merged**